### PR TITLE
Ar/cpp runner

### PR DESCRIPTION
**Pull request by ana-roman**
Implemented a new runner that will use Arcan to analyze C projects. 
Also added the possibility for both Java and C projects to analyze only one version instead of the whole project.

### CODE REVIEW

File: src/main/java/org/rug/data/project/GitVersion.java [hunk 1]
```diff
@@ -49,10 +51,14 @@ public synchronized SourceCodeRetriever getSourceCodeRetriever() {
...
         var fileName = f.getFileName().toString();
         int endIndex = f.toFile().isDirectory() ? fileName.length() : fileName.lastIndexOf('.');
-        var splits = fileName.substring(0, endIndex).split("-");
-        setVersionIndex(Long.parseLong(splits[1]));
-        versionDate = String.join("-", splits[2].split("_"));
-        commitName = splits[3];
+        var splits = new ArrayList<>(Arrays.asList(fileName.substring(0, endIndex).split("-")));
+        if (splits.get(1) .equals("file")) {
+            // Arcan for singleVersion Java adds the word "file" to the outputted GraphML file
+            splits.remove(1);
```
Comment (darius-sas) [inline src/main/java/org/rug/data/project/GitVersion.java]: This check shouldn't be necessary, as I changed arcan to output the file names consistently across single-version and history analysis modes.

File: src/main/java/org/rug/args/ProjectArgsManager.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
     public File gitRepo;
 
     @Parameter(names = {"-javaProject", "-jP"}, description = "Flag this as a Java project (i.e. project analysed with Arcan for Java).")
-    public boolean isJava = true;
```
Comment (darius-sas) [inline src/main/java/org/rug/args/ProjectArgsManager.java:21]: Why did you change this?  
I'm asking because the analysis for Java is supposed to be the default one. If you change this here, the user has to use -jP every time he analyses a java project. 
Unless this is really necessary, I would suggest to undo this change.

**Review by darius-sas — APPROVED** [review]

File: src/main/java/org/rug/args/ProjectArgsManager.java (hunk 2 above)
Comment (ana-roman) [reply src/main/java/org/rug/args/ProjectArgsManager.java:21]: As discussed, this was messing with the C runner since the isJava flag was set to true by default and it would still be true when running  a C project. For the web API, the user has to set the `language` parameter anyway!

File: src/main/java/org/rug/data/project/GitVersion.java (hunk 1 above)
Comment (ana-roman) [reply src/main/java/org/rug/data/project/GitVersion.java]: Rolled back the changes!

**Comment by ana-roman** [comment]
Updated the code with the review comments and fixed the conflict.

**Comment by darius-sas** [comment]
Good, can I merge now?

**Comment by ana-roman** [comment]
Should be ready for takeoff! 🚀
